### PR TITLE
[v1.1.1] Extract WeatherForecastItem component, extract functions when needed

### PR DESCRIPTION
**Pull request by nktssh**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,21 +56,18 @@ function isCurrentCityDataLoaded() {
...
-renderApp();
-
 function reRenderApp() {
   // Need to split rendering cycles
-  setTimeout(() => window.renderApp(), 0);
+  setTimeout(window.renderApp, 0);
 }
 
-function loadData() {
+function validateAndLoadData() {
```
Comment (OleksiyRudenko) [inline index.js:64]: If `validateAndLoadData` was fully thenable (i.e. returned a promise, resolved in some cases), then we could fire re-render beyond it.

This would also enable network errors handling.

File: index.js [hunk 2]
```diff
@@ -45,21 +56,18 @@ function isCurrentCityDataLoaded() {
   return cityByWeather.hasOwnProperty(currentCity);
 }
 
-renderApp();
-
 function reRenderApp() {
   // Need to split rendering cycles
-  setTimeout(() => window.renderApp(), 0);
+  setTimeout(window.renderApp, 0);
```
Comment (OleksiyRudenko) [inline index.js:61]: Why is that required?

A side note: if there is a delay assigned anyway, perhaps we could use a variable delay to emulate network requests that take longer when re-render gets fired on actual data load from remote server.

**Review by OleksiyRudenko — COMMENTED** [review]
@nktssh please check the comments above

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "frontend-2021-framework",
-  "version": "1.1.0",
+  "version": "1.1.1",
```
Comment (zonzujiro) [inline package.json:3]: ❤️

File: index.js (hunk 2 above)
Comment (zonzujiro) [inline index.js:61]: Can you please describe why this is necessary?

When I wrote my version of the framework, I have used `setTimeout` to re-render the app after the event, just to give a time for event bubbling, etc.

This is a case? :)

Comment (nktssh) [reply index.js:61]: For now it's because of implementation - had to split the cycles because we assign all the content to the node, and changing vars in the process makes data shown outdated.
Should not be the case with promise-based data loading, will check that case.

Comment (nktssh) [reply index.js:61]: Regarding splitting rendering - see [this thread ](https://github.com/kottans/framework-2021/pull/20#discussion_r607600328)

We could use dev tools to emulate network delays etc. Don't want to make it overcomplicated.

Comment (zonzujiro) [reply index.js:61]: Can you please leave a comment? Cause this is a little bit unclear from the outside.

Comment (nktssh) [reply index.js]: No comments needed, promise-base API resolved this case. Thanks for the notice! 🙏

File: index.js [hunk 4]
```diff
@@ -21,18 +21,36 @@ window.dataStore = window.dataStore || {
...
+      window.dataStore.isDataLoading = false;
+
+      if (error) {
+        window.dataStore.error = error;
+      } else if (data) {
+        window.dataStore.cityByWeather[cityName] = data;
+      }
+    })
+    .catch(() => {
+      window.dataStore.error = 'Some error occurred.';
```
Comment (nktssh) [inline index.js:46]: @OleksiyRudenko this is the place where we handle network errors if any

**Comment by nktssh** [comment]
@OleksiyRudenko @zonzujiro 
updated, please take a look when you'll have time.

**Review by zonzujiro — APPROVED** [review]

**Review by OleksiyRudenko — APPROVED** [review]
